plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+
+    /**
+     * Convert response body to Metric object.
+     *
+     * @param deployResource The deployed resource.
+     * @param metricName     metricName.
+     * @param result         result.
+     * @return Returns Metric object.
+     */
+    public static Metric convertResponseToMetric(DeployResource deployResource,
+                                                 String metricName, String result) {
+        Metric metric = new Metric();
+        metric.setName(metricName);
+        Map<String, String> labels = new HashMap<>();
+        labels.put("id", deployResource.getResourceId());
+        labels.put("name", deployResource.getName());
+        metric.setLabels(labels);
+        metric.setType(MetricType.GAUGE);
+        try {
+            JsonNode resultJson = objectMapper.readTree(result);
+            if (Objects.nonNull(resultJson) && Objects.nonNull(resultJson.get("datapoints"))) {
+                List<Datapoint> datapointList =
+                        objectMapper.readValue(resultJson.get("datapoints").toString(),
+                                new TypeReference<>() {
+                                });
+                if (!CollectionUtils.isEmpty(datapointList)) {
+                    datapointList.sort(Comparator.comparing(Datapoint::getTimestamp).reversed());
+                    MetricItem metricItem = new MetricItem();
+                    metricItem.setType(MetricItemType.VALUE);
+                    metricItem.setValue(datapointList.get(0).getAverage());
+                    metric.setMetrics(List.of(metricItem));
+                }
+            }
+        } catch (IOException e) {
+            log.error("convertResponseToMetric error.", e);
+        }
+        return metric;
+    }
+
+    /**
+     * Get id of project from response body.
+     *
+     * @param result result.
+     * @return Returns id of project.
+     */
+    public static String getProjectIdFromResponse(String result) {
+        String projectId = "";
+        try {
+            JsonNode resultJson = objectMapper.readTree(result);
+            if (Objects.nonNull(resultJson) && Objects.nonNull(resultJson.get("projects"))) {
+                JsonNode projects = resultJson.get("projects");
+                if (projects.isArray() && !projects.isEmpty()) {
+                    projectId = projects.get(0).get("id").toString().replaceAll("\"", "");
+                }
+            }
+        } catch (IOException e) {
+            log.error("getProjectIdFromResponse error.", e);
+        }
+        return projectId;
+    }
+
+    private static StringBuilder getQueryMetricBasicUrl(String region, String projectId) {
+        return new StringBuilder(FlexibleEngineMonitorConstants.PROTOCOL_HTTPS)
+                .append(FlexibleEngineMonitorConstants.CES_ENDPOINT_PREFIX)
+                .append(region)
+                .append(FlexibleEngineMonitorConstants.ENDPOINT_SUFFIX).append("/")
+                .append("V1.0").append("/")
```

Comment:
the version can also be moved to constants file.

Reply:
OK.